modules/base/tool.go

```diff
@@ -206,11 +206,26 @@ func HashEmail(email string) string {
 
 // AvatarLink returns avatar link by given email.
 func AvatarLink(email string) string {
-	if setting.DisableGravatar || setting.OfflineMode {
-		return setting.AppSubUrl + "/img/avatar_default.png"
+	var url string
+
+	if !setting.OfflineMode {
+		if setting.EnableFederatedAvatar && setting.LibravatarService != nil {
```

Comment:
I agree. Note that I didn't change that to keep focused on the topic of this PR. What you suggest is also valid in current develop branch (no need to check both OfflineMode and DisableGravatar). How about fixing that in develop so that I then clean it up here on next rebase ?